embedding/csharp/Tizen.Flutter.Embedding/NUIFlutterView.cs

```diff
@@ -0,0 +1,205 @@
+
+            if (Engine == null)
+            {
+                Engine = new FlutterEngine();
+            }
+
+            if (!Engine.IsValid)
+            {
+                TizenLog.Error("Could not create a Flutter engine.");
+                return false;
+            }
+
+            _viewSize = GetDefaultSize();
+
+            Type baseType = typeof(NativeImageQueue).BaseType.BaseType.BaseType;
+            FieldInfo field = baseType.GetField("swigCPtr", BindingFlags.NonPublic | BindingFlags.Instance);
+            NativeImageQueue nativeImageQueue =
+                new NativeImageQueue((uint)_viewSize.Width, (uint)_viewSize.Height, NativeImageQueue.ColorFormat.RGBA8888);
+            HandleRef nativeImageQueueRef = (HandleRef)field.GetValue(nativeImageQueue);
+            SetImage(nativeImageQueue.GenerateUrl().ToString());
+
+            Type imageViewBaseType = typeof(ImageView).BaseType.BaseType.BaseType.BaseType;
+            FieldInfo imageViewField =
+                imageViewBaseType.GetField("swigCPtr", BindingFlags.NonPublic | BindingFlags.Instance);
+            HandleRef imageViewRef = (HandleRef)imageViewField.GetValue(this);
+
+            var viewProperties = new FlutterDesktopViewProperties
+            {
+                width = _viewSize.Width,
+                height = _viewSize.Height,
+            };
+
+            View = FlutterDesktopViewCreateFromImageView(
+                ref viewProperties, Engine.Engine, imageViewRef.Handle, nativeImageQueueRef.Handle,
+                NUIApplication.GetDefaultWindow().GetNativeId());
+            if (View.IsInvalid)
+            {
+                TizenLog.Error("Could not launch a Flutter view.");
+                return false;
+            }
+
+            RegisterEvent();
+            return true;
+        }
+
+        /// <summary>
+        /// Terminates the running view and the associated engine.
+        /// </summary>
+        public void Destroy()
+        {
+            if (IsRunning)
+            {
+                FlutterDesktopViewDestroy(View);
+                Engine = null;
+                View = new FlutterDesktopView();
+            }
+        }
+
+        /// <summary>
+        /// Returns the size that can be the default.
+        /// </summary>
+        private Size2D GetDefaultSize()
+        {
+            if (Size2D.Width == 0 || Size2D.Height == 0)
+            {
+                View parent = GetParent() as View;
+                if (!parent)
+                {
+                    Window window = NUIApplication.GetDefaultWindow();
+                    return new Size2D(window.Size.Width, window.Size.Height);
+                }
+                return parent.Size2D;
+            }
+            return Size2D;
+        }
+
+        /// <summary>
+        /// Register view event callbacks.
+        /// </summary>
+        private void RegisterEvent()
+        {
+            Focusable = true;
+            KeyEvent += (object s, KeyEventArgs e) =>
```

Comment:
You might have to unregister the handlers as well.

Reply:
unregister is unnecessary